File: src/objc/src/HproseException.m
/**********************************************************\
|                                                          |
|                          hprose                          |
|                                                          |
| Official WebSite: http://www.hprose.com/                 |
|                   http://www.hprose.net/                 |
|                   http://www.hprose.org/                 |
|                                                          |
\**********************************************************/
/**********************************************************\
 *                                                        *
 * HproseException.m                                      *
 *                                                        *
 * hprose exception class for Objective-C.                *
 *                                                        *
 * LastModified: May 16, 2010                             *
 * Author: Ma Bingyao <andot@hprfc.com>                   *
 *                                                        *
\**********************************************************/

#import "HproseException.h"

@implementation HproseException

+ (HproseException *)exceptionWithReason:(NSString *)reason {
    return (HproseException *)[HproseException
        exceptionWithName:@"HproseException"
        reason:reason
        userInfo:nil];
}

@end

File: src/dotnet/src/Hprose/Common/HproseException.cs
/**********************************************************\
|                                                          |
|                          hprose                          |
|                                                          |
| Official WebSite: http://www.hprose.com/                 |
|                   http://www.hprose.net/                 |
|                   http://www.hprose.org/                 |
|                                                          |
\**********************************************************/
/**********************************************************\
 *                                                        *
 * HproseException.cs                                     *
 *                                                        *
 * hprose exception class for C#.                         *
 *                                                        *
 * LastModified: May 16, 2010                             *
 * Author: Ma Bingyao <andot@hprose.com>                  *
 *                                                        *
\**********************************************************/
using System;
using System.IO;

namespace Hprose.Common {
    public class HproseException : IOException {
        public HproseException()
            : base() {
        }
        public HproseException(string message)
            : base(message) {
        }
        public HproseException(string message, Exception innerException)
            : base(message, innerException) {
        }
    }
}